core/backend/src/Element.ts

```diff
@@ -372,25 +365,24 @@ export class Element extends Entity {
     return this.getReferenceIds();
   }
 
-  /** Get the Ids of this element's *references*. A *reference* is an element that this element references.
-   * This is important for cloning operations but can be useful in other situations as well.
-   * @see collectReferenceIds
-   * @beta
-   */
-  public getReferenceIds(): Id64Set {
-    const referenceIds = new Set<Id64String>();
-    this.collectReferenceIds(referenceIds);
-    return referenceIds;
-  }
-
+  // FIXME: since this is beta, replace it with an array of objects containing the reference type
   /** A *required reference* is an element that had to be inserted before this element could have been inserted.
    * This is the list of property keys on this element that store references to those elements
-   * @note This should be overridden (with `super` called) at each level the class hierarchy that introduces required references.
+   * @note This should be overridden (with `super` called) at each level of the class hierarchy that introduces required references.
    * @note any property listed here must be added to the reference ids in [[collectReferenceIds]]
    * @beta
    */
   public static readonly requiredReferenceKeys: ReadonlyArray<string> = ["parent", "model"];
 
+  /** A map of every [[requiredReferenceKeys]] on this class to their entity type.
+   * @note This should be overridden (with `super` called) at each level of the class hierarchy that introduces required references.
+   * @beta
+   */
+  public static readonly requiredReferenceKeyTypeMap: Record<string, ConcreteEntityTypes> = {
```

Comment:
alpha beta please grep and find all occurrences.

Reply:
done